gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/state/FileTokenStateJournal.java

```diff
@@ -0,0 +1,215 @@
+     *
+     * @param entry A journal file entry line
+     *
+     * @return A FileJournalEntry object created from the specified entry.
+     */
+    protected FileJournalEntry parse(final String entry) {
+        return FileJournalEntry.parse(entry);
+    }
+
+    /**
+     * A JournalEntry implementation for File-based TokenStateJournal implementations
+     */
+    static final class FileJournalEntry implements JournalEntry {
+        private final String tokenId;
+        private final String issueTime;
+        private final String expiration;
+        private final String maxLifetime;
+
+        FileJournalEntry(final String tokenId, long issueTime, long expiration, long maxLifetime) {
+            this(tokenId, String.valueOf(issueTime), String.valueOf(expiration), String.valueOf(maxLifetime));
+        }
+
+        FileJournalEntry(final String tokenId,
+                         final String issueTime,
+                         final String expiration,
+                         final String maxLifetime) {
+            this.tokenId = tokenId;
+            this.issueTime = String.valueOf(issueTime);
+            this.expiration = String.valueOf(expiration);
+            this.maxLifetime = String.valueOf(maxLifetime);
+        }
+
+        @Override
+        public String getTokenId() {
+            return tokenId;
+        }
+
+        @Override
+        public String getIssueTime() {
+            return issueTime;
+        }
+
+        @Override
+        public String getExpiration() {
+            return expiration;
+        }
+
+        @Override
+        public String getMaxLifetime() {
+            return maxLifetime;
+        }
+
+        @Override
+        public String toString() {
+            String[] elements = new String[4];
+
+            elements[INDEX_TOKEN_ID] = getTokenId();
+
+            String issueTime = getIssueTime();
+            elements[INDEX_ISSUE_TIME] = (issueTime != null) ? issueTime : "";
+
+            String expiration = getExpiration();
+            elements[INDEX_EXPIRATION] = (expiration != null) ? expiration : "";
+
+            String maxLifetime = getMaxLifetime();
+            elements[INDEX_MAX_LIFETIME] = (maxLifetime != null) ? maxLifetime : "";
+
+            return String.format(Locale.ROOT,
+                                 "%s,%s,%s,%s",
+                                 elements[INDEX_TOKEN_ID],
+                                 elements[INDEX_ISSUE_TIME],
+                                 elements[INDEX_EXPIRATION],
+                                 elements[INDEX_MAX_LIFETIME]);
+        }
+
+        /**
+          * Parse the String representation of an entry.
+          *
+          * @param entry A journal file entry line
+          *
+          * @return A FileJournalEntry object created from the specified entry.
+          */
+        static FileJournalEntry parse(final String entry) {
+            String[] elements = entry.split(",");
```

Comment:
I feel the ',' (the colon) should be a constant indicating the SEPARATOR and used within `toString()` above too.